rust/connlib/tunnel/src/dns.rs

```diff
@@ -217,48 +216,44 @@ impl StubResolver {
         ips
     }
 
-    // This function will panic if it's called with an invalid PTR question
-    fn get_records<N: ToName>(&mut self, question: &Question<N>) -> Vec<RecordData<DomainName>> {
+    fn get_records<N: ToName>(
+        &mut self,
+        question: &Question<N>,
+    ) -> Option<Vec<RecordData<DomainName>>> {
         match question.qtype() {
-            Rtype::A => self
-                .get_or_assign_ips(question.qname().to_name())
-                .iter()
-                .copied()
-                .filter_map(get_v4)
-                .map(domain::rdata::A::new)
-                .map(RecordData::A)
-                .collect_vec(),
-
-            Rtype::AAAA => self
-                .get_or_assign_ips(question.qname().to_name())
-                .iter()
-                .copied()
-                .filter_map(get_v6)
-                .map(domain::rdata::Aaaa::new)
-                .map(RecordData::Aaaa)
-                .collect_vec(),
-            Rtype::PTR => {
-                let Some(ip) = reverse_dns_addr(&question.qname().to_name::<Vec<_>>().to_string())
-                else {
-                    return Vec::new();
-                };
-                let Some(fqdn) = self.ips_to_fqdn.get(&ip) else {
-                    debug_assert!(false, "we expect this function to be called only with PTR records for resource ips");
-                    return Vec::new();
-                };
-
-                vec![RecordData::Ptr(domain::rdata::Ptr::new(fqdn.clone()))]
+            Rtype::A => {
+                get_description(&question.qname().to_name(), &self.dns_resources)?;
+
+                Some(
+                    self.get_or_assign_ips(question.qname().to_name())
+                        .iter()
+                        .copied()
+                        .filter_map(get_v4)
+                        .map(domain::rdata::A::new)
+                        .map(RecordData::A)
+                        .collect_vec(),
+                )
             }
-            _ => Vec::new(),
-        }
-    }
+            Rtype::AAAA => {
+                get_description(&question.qname().to_name(), &self.dns_resources)?;
+
+                Some(
+                    self.get_or_assign_ips(question.qname().to_name())
+                        .iter()
+                        .copied()
+                        .filter_map(get_v4)
+                        .map(domain::rdata::A::new)
+                        .map(RecordData::A)
```

Comment:
:facepalm:

Reply:
Another victory for the proptests! haha @thomaseizinger 

Though, we might want to consider a cleaner output. What I see in the logs isn't very clear https://github.com/firezone/firezone/actions/runs/9752948976/job/26917419077?pr=5674#step:5:7357

Reply:
I think it is one of those where looking a bit further up to a previous iteration (i.e. before the shrinking) helps!

Reply:
> Another victory for the proptests! haha @thomaseizinger

:rocket: :rocket: :rocket: